test/e2e/azure_logcollector.go

```diff
@@ -416,16 +415,17 @@ func collectVMBootLog(ctx context.Context, am *infrav1.AzureMachine, outputPath
 		return errors.Wrap(err, "failed to parse resource id")
 	}
 
-	settings, err := auth.GetSettingsFromEnvironment()
-	if err != nil {
-		return errors.Wrap(err, "failed to get settings from environment")
+	subscriptionID := os.Getenv("AZURE_SUBSCRIPTION_ID")
+	if subscriptionID == "" {
+		return errors.New("AZURE_SUBSCRIPTION_ID is not set")
```

Comment:
I wonder if it might be worth having two overlapping flavors of the helper, where one looks like what happens here:
```go
func getSubAndError() (string, error) {...}
```

and a separate one that asserts the above err is nil:
```go
func assertSub() string {
	sub, err := getSubAndErr()
	Expect(err).NotTo(HaveOccurred())
	return sub
}
```

(but with better names)

Reply:
In all the cases where this is called, it seems to me it will be an error if `AZURE_SUBSCRIPTION_ID == ""`. And there are no other errors to pass on or to create. So I'm not sure why we need both versions; I just went for the equivalent of `assertSub()`.